src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py

```diff
@@ -142,7 +248,15 @@ def evaluate(
         )
 
     input_data_df = pd.read_json(data, lines=True)
-    input_data_df = input_data_df.rename(columns={col: f"inputs.{col}" for col in input_data_df.columns})
+    if tempfile_created:
+        # During the run we have created the temporary file. We will delete it here.
+        os.unlink(data)
+
+    # Rename columns, generated by template function to outputs instead of inputs.
+    template_generated_columns = set(input_data_df) - set(initial_columns)
+    input_data_df.rename(columns={
+        col: f"{'outputs' if col in template_generated_columns else 'inputs'}.{col}" for col in input_data_df.columns},
+        inplace=True)

```

Comment:
what's template_generated_columns? don't quite understand the logic here.

Reply:
It should be target_generated_columns essentially we do not know, which columns are generated by target function and hence, should be renamed. Here we are comparing column set with the original data frame and adding "inputs" prefix to preexisting columns and "outputs" to the ones, we have created by target function.